### ISSUE TITLE
Querying empty sub-documents returns all documents

### ISSUE THREAD

**Issue by petee-d** [mongomock/mongomock#341]
```
    def test__find_by_empty_document(self):
        self.cmp.do.insert({"doc": {"data": "val"}})
        self.cmp.do.insert({"doc": {}})
        self.cmp.do.insert({"doc": None})
        self.cmp.compare.find({"doc": {}})
```
In mongo this should only return documents where the value is an empty sub-document, mongomock instead matches everything thinking the field has no constraints. I will do a pull request with the fix shortly.

**Comment by petee-d** [comment]
This still isn't fixed and it looks like my PR can still be merged without conflicts. Could someone please review or implement a better fix?

**Comment by vmalloc** [comment]
@petee-d deepest apologies. It took me a long time to go over the backlog and somehow I missed that one.

Thanks for your contribution!

**Comment by petee-d** [comment]
@vmalloc And thank you very much for your valuable time spent at this great library. I'm very happy to see that since I'm now able to use the latest code instead of my fork, many tests in our project that previously didn't work with mongomock now pass. :+1: 

By the way, most of the failing ones are because of this: https://github.com/mongomock/mongomock/issues/385 (dots in projection), maybe I'll submit a PR for that once I have more time.

**Comment by vmalloc** [comment]
@petee-d thanks for the feedback! Any help is appreciated -- I'm not actively using Mongo these days so I have less time to maintain mongomock.

**Comment by pcorpet** [comment]
This issue was fixed by #342.
